src/bridge/IrcHandler.ts

```diff
@@ -793,20 +793,28 @@ export class IrcHandler {
         }
 
         // get virtual matrix user
-        req.log.info("Mapped nick %s to %s", nick, userId);
-        const promise = await Promise.all(matrixRooms.map(async (room) => {
-            await this.membershipQueue.leave(
-                room.getId(), userId, req, true, "Client PARTed from channel",
-                leavingUser.isVirtual ? this.ircBridge.appServiceUserId : undefined);
-            try {
-                await this.roomAccessSyncer.removePowerLevels(room.getId(), [userId]);
+        req.log.info("Mapped nick %s to %s (leaving %s room(s))", nick, userId, matrixRooms.length);
+        await Promise.all(matrixRooms.map(async (room) => {
+            if (leavingUser.isVirtual) {
+                return this.membershipQueue.leave(
```

Comment:
A quick read of https://github.com/matrix-org/matrix-appservice-bridge/blob/develop/src/components/membership-queue.ts didn't enlighten me - how are errors handled here? This `Promise.all` doesn't seem to be stored anywhere, we just await and ignore the return value?

Reply:
membershipQueue.(join|leave) will return a promise that will eventually resolve or reject depending on if we successfully send presence (after a set number of retries). If any of these fail, the promise.all will fail, and that will be caught in the `complete` function in `IrcEventBroker`, which in turn logs an error and that's it.

Reply:
great summary thanks